aquadoggo/src/db/stores/query.rs

```diff
@@ -0,0 +1,1703 @@
+        // Otherwise just query the documents directly
+        None => r#"
+            documents
+            JOIN document_view_fields
+                ON documents.document_view_id = document_view_fields.document_view_id
+        "#
+        .to_string(),
+    }
+}
+
+fn from_fields_sql(fields: &ApplicationFields) -> String {
+    if fields.is_empty() {
+        ""
+    } else {
+        r#"
+        JOIN operation_fields_v1
+            ON
+                document_view_fields.operation_id = operation_fields_v1.operation_id
+                AND
+                    document_view_fields.name = operation_fields_v1.name
+        "#
+    }
+    .to_string()
+}
+
+impl SqlStore {
+    /// Returns a paginated collection of documents from the database which can be filtered and
+    /// ordered by custom parameters.
+    ///
+    /// When passing a `list` configuration the query will run against the documents of a (pinned)
+    /// relation list instead.
+    pub async fn query(
+        &self,
+        schema: &Schema,
+        args: &Query<DocumentCursor>,
+        list: Option<&RelationList>,
+    ) -> Result<QueryResponse, DocumentStorageError> {
+        let schema_id = schema.id();
+
+        // Get all selected application fields from query
+        let application_fields = args.select.application_fields();
+
+        // Generate SQL based on the given schema and query
+        let select_edited = select_edited_sql(&args.select);
+        let select_owner = select_owner_sql(&args.select);
+        let select_fields = select_fields_sql(&application_fields, list);
+
+        let from = from_sql(list);
+        let from_fields = from_fields_sql(&application_fields);
+
+        let where_ = where_sql(schema.id(), &application_fields, list);
+        let and_fields = where_fields_sql(&application_fields);
+        let and_filters = where_filter_sql(&args.filter, schema);
+        let and_pagination = where_pagination_sql(&args.pagination, list, &args.order);
+
+        let group = group_sql(&application_fields);
+        let order = order_sql(&args.order, schema, list);
+        let (page_size, limit) = limit_sql(&args.pagination, &application_fields);
+
+        let sea_quel = format!(
+            r#"
+            -- ░░░░░▄▀▀▀▄░░░░░░░░░░░░░░░░░
+            -- ▄███▀░◐░░░▌░░░░░░░░░░░░░░░░
+            -- ░░░░▌░░░░░▐░░░░░░░░░░░░░░░░
+            -- ░░░░▐░░░░░▐░░░░░░░░░░░░░░░░
+            -- ░░░░▌░░░░░▐▄▄░░░░░░░░░░░░░░
+            -- ░░░░▌░░░░▄▀▒▒▀▀▀▀▄░░░░░░░░░
+            -- ░░░▐░░░░▐▒▒▒▒▒▒▒▒▀▀▄░░░░░░░
+            -- ░░░▐░░░░▐▄▒▒▒▒▒▒▒▒▒▒▀▄░░░░░
+            -- ░░░░▀▄░░░░▀▄▒▒▒▒▒▒▒▒▒▒▀▄░░░
+            -- ░░░░░░▀▄▄▄▄▄█▄▄▄▄▄▄▄▄▄▄▄▀▄░
+            -- ░░░░░░░░░░░▌▌░▌▌░░░░░░░░░░░
+            -- ░░░░░░░░░░░▌▌░▌▌░░░░░░░░░░░
+            -- ░░░░░░░░░▄▄▌▌▄▌▌░░░░░░░░░░░
+            --
+            -- Hello, my name is sea gull.
+            -- I designed a SQL query which
+            -- is as smart, as big and as
+            -- annoying as myself.
```

Comment:
What can I say. I just love our new SQL Sea Gull friend.

I see you already refactored the query generation methods out, that's cool, it's easy to read through the different steps taken in the query now.

Reply:
In the meeting I've mentioned to bring them into separate modules, they were already summarized like this inside the query.